lms/envs/bok_choy_docker.auth.json

```diff
@@ -0,0 +1,148 @@
+    "CELERY_BROKER_USER": "celery",
+    "CONTENTSTORE": {
+        "DOC_STORE_CONFIG": {
+            "collection": "modulestore",
+            "db": "test",
+            "host": [
+                "edx.devstack.mongo"
+            ],
+            "port": 27017
+        },
+        "ENGINE": "xmodule.contentstore.mongo.MongoContentStore",
+        "OPTIONS": {
+            "db": "test",
+            "host": [
+                "edx.devstack.mongo"
+            ],
+            "port": 27017
+        }
+    },
+    "DATABASES": {
+        "default": {
+            "ENGINE": "django.db.backends.mysql",
+            "HOST": "edx.devstack.mysql",
+            "NAME": "edxtest",
+            "PASSWORD": "",
+            "PORT": "3306",
+            "USER": "root"
+        },
+        "student_module_history": {
+            "ENGINE": "django.db.backends.mysql",
+            "HOST": "edx.devstack.mysql",
+            "NAME": "student_module_history_test",
+            "PASSWORD": "",
+            "PORT": "3306",
+            "USER": "root"
+        }
+    },
+    "DOC_STORE_CONFIG": {
+        "collection": "modulestore",
+        "db": "test",
+        "host": [
+            "edx.devstack.mongo"
+        ],
+        "port": 27017
+    },
+    "TRACKING_BACKENDS": {
+        "mongo": {
+            "ENGINE": "track.backends.mongodb.MongoBackend",
+            "OPTIONS": {
+                "database": "test",
+                "collection": "events",
+                "host": [
+                    "edx.devstack.mongo"
+                ],
+                "port": 27017
+            }
+        }
+    },
+    "EVENT_TRACKING_BACKENDS": {
+        "mongo": {
+            "ENGINE": "eventtracking.backends.mongodb.MongoBackend",
+            "OPTIONS": {
+                "database": "test",
+                "collection": "events",
+                "host": [
+                    "edx.devstack.mongo"
+                ],
+                "port": 27017
+            }
+        }
+    },
+    "MODULESTORE": {
+        "default": {
+            "ENGINE": "xmodule.modulestore.mixed.MixedModuleStore",
+            "OPTIONS": {
+                "mappings": {},
+                "stores": [
+                    {
+                        "NAME": "draft",
+                        "DOC_STORE_CONFIG": {
+                            "collection": "modulestore",
+                            "db": "test",
+                            "host": [
+                                "edx.devstack.mongo"
+                            ],
+                            "port": 27017
+                        },
+                        "ENGINE": "xmodule.modulestore.mongo.DraftMongoModuleStore",
+                        "OPTIONS": {
+                            "collection": "modulestore",
+                            "db": "test",
+                            "default_class": "xmodule.hidden_module.HiddenDescriptor",
+                            "fs_root": "** OVERRIDDEN **",
```

Comment:
I originally tried just overriding the specific settings which needed to change via bok_choy_docker.py instead of copying the JSON file, but:

* The settings involved were pretty deep in nested dictionaries and arrays, making it hard to get the overrides correct
* You couldn't see the overrides in context of the related settings
* Most importantly, some code was being initialized in the base settings file with the original values before the overrides could be applied, which caused server startup to fail

I think refactoring this to reduce duplication will have to happen as part of a larger settings overhaul.